rust/lance-encoding/src/encodings/physical/bitpack_fastlanes.rs

```diff
@@ -1560,3 +1564,251 @@ mod tests {
     //     check_round_trip_bitpacked(arr).await;
     // }
 }
+
+// This macro chunks the FixedWidth DataBlock, bitpacks them with 1024 values per chunk,
+// it puts the bit-width parameter in front of each chunk,
+// and the bit-width parameter has the same bit-width as the uncompressed DataBlock
+// for example, if the input DataBlock has `bits_per_value` of `16`, there will be 2 bytes(16 bits)
+// in front of each chunk storing the `bit-width` parameter.
+macro_rules! chunk_data_impl {
+    ($data:expr, $data_type:ty) => {{
+        let data_buffer = $data.data.borrow_to_typed_slice::<$data_type>();
+        let data_buffer = data_buffer.as_ref();
+
+        let bit_widths = $data
+            .get_stat(Stat::BitWidth)
+            .expect("FixedWidthDataBlock should have valid bit width statistics");
+        println!("bit_widths statistics got");
+        let bit_widths_array = bit_widths
+            .as_any()
+            .downcast_ref::<PrimitiveArray<UInt64Type>>()
+            .unwrap();
+
+        let (packed_chunk_sizes, total_size) = bit_widths_array
+            .values()
+            .iter()
+            .map(|&bit_width| {
+                let chunk_size = ((1024 * bit_width) / $data.bits_per_value) as usize;
+                (chunk_size, chunk_size + 1)
+            })
+            .fold(
+                (Vec::with_capacity(bit_widths_array.len()), 0),
+                |(mut sizes, total), (size, inc)| {
+                    sizes.push(size);
+                    (sizes, total + inc)
+                },
+            );
+
+        let mut output: Vec<$data_type> = Vec::with_capacity(total_size);
+        let mut chunks = Vec::with_capacity(bit_widths_array.len());
+
+        for i in 0..bit_widths_array.len() - 1 {
+            let start_elem = i * ELEMS_PER_CHUNK as usize;
+            let bit_width = bit_widths_array.value(i) as $data_type;
+            output.push(bit_width);
+            let output_len = output.len();
+            unsafe {
+                output.set_len(output_len + packed_chunk_sizes[i]);
+                BitPacking::unchecked_pack(
+                    bit_width as usize,
+                    &data_buffer[start_elem..][..ELEMS_PER_CHUNK as usize],
+                    &mut output[output_len..][..packed_chunk_sizes[i]],
+                );
+            }
+            chunks.push(MiniBlockChunk {
+                num_bytes: ((1 + packed_chunk_sizes[i]) * std::mem::size_of::<$data_type>()) as u16,
+                log_num_values: 10,
```

Comment:
Since `ELEMS_PER_CHUNK` is a constant it would be good for this to be a constant too.  Maybe just `LOG_ELEMENTS_PER_CHUNK`.

Reply:
fixed, thanks!